File: algorithms/php/1.TwoSum.php
<?php

class Solution {

    /**
     * @param Integer[] $nums
     * @param Integer $target
     * @return Integer[]
     */
    function twoSum($nums, $target) {
        $count = count($nums);

        for ($i = 0; $i < $count; $i++) {
            for ($j = $i + 1; $j < $count; $j++) {
                if ($nums[$i] + $nums[$j] === $target) {
                    return [$i, $j];
                }
            }
        }
    }
}

class Solution_advanced {

    /**
     * @param Integer[] $nums
     * @param Integer $target
     * @return Integer[]
     */
    function twoSum($nums, $target) {
        $keys = [];
        $count = count($nums);

        for ($i = 0; $i < $count; $i++) {
            $current = $target - $nums[$i];
            if (!array_key_exists($current, $keys)) {
                $keys[$nums[$i]] = 1;
            } else {
                return [$i, array_search($current, $nums)];
            }
        }

        return [];
    }
}

File: algorithms/cpp/1.TwoSum.cpp
#include "vector"
#include "set"
#include "algorithm"
#include "functions.h"

using namespace std;

class Solution {
public:
    vector<int> twoSum(vector<int> &nums, int target) {
        set<int> keys;
        int n = nums.size();

        for (int i = 0; i < n; i++) {
            int current = target - nums[i];
            if (keys.find(current) == keys.end()) {
                keys.insert(nums[i]);
            } else {
                int second_index = find(nums.begin(), nums.end(), current) - nums.begin();
                return vector<int>{i, second_index};
            }
        }

        return vector<int>{};
    }
};

int main() {
    Solution s;
    vector<int> in{3, 2, 4};
    vector<int> out = s.twoSum(in, 6);
    print_vector(out);
}